flink-metrics/flink-metrics-otel/src/main/java/org/apache/flink/traces/otel/OpenTelemetryTraceReporter.java

```diff
@@ -70,10 +70,14 @@ public void open(MetricConfig metricConfig) {
 
     @Override
     public void close() {
-        spanProcessor.forceFlush();
-        spanProcessor.close();
-        spanExporter.flush();
-        spanExporter.close();
+        if (spanProcessor != null) {
```

Comment:
I know this is a back port, I  am wondering if we get an exception processing the spanProcessor shouldn't we still try to clean up spanExporter if we have one. i.e. put the spanExporter code in a finally.